functionary/ui/views/environments/create.py

```diff
@@ -0,0 +1,96 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.core.exceptions import ValidationError
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect, QueryDict
+from django.shortcuts import get_object_or_404, render
+from django.urls import reverse
+from django.views.decorators.http import require_GET
+from django.views.generic import View
+from thefuzz import process
+
+from core.auth import Permission
+from core.models import Environment, EnvironmentUserRole, User
+from ui.forms.environments import EnvForm
+
+
+class EnvironmentCreateMemberView(LoginRequiredMixin, UserPassesTestMixin, View):
+    def get(self, request: HttpRequest, environment_id: str):
+        environment = get_object_or_404(Environment, id=environment_id)
+
+        form = EnvForm()
+        context = {
+            "form": form,
+            "environment_id": str(environment.id),
+        }
+        return render(
+            request, "partials/environments/environment_add_user.html", context
+        )
+
+    def post(self, request: HttpRequest, environment_id: str):
+        """Add user to the environment"""
+        environment = get_object_or_404(Environment, id=environment_id)
+
+        data: QueryDict = request.POST.copy()
+        user = User.objects.filter(username=data["user"]).first()
+        data["environment"] = environment
+        data["user"] = user
+
+        form = EnvForm(data=data)
+        if not form.is_valid():
+            if user is None:
+                form.add_error(
+                    "user",
+                    ValidationError("User not found.", code="invalid"),
+                )
+
+            context = {"form": form, "environment_id": str(environment.id)}
+            return render(
+                request, "forms/environments/environment_add_user.html", context
+            )
+
+        """
+        This endpoint is for adding a user to the environment,
+        not updating a user's role on the environment
+        """
+        if (
+            _ := EnvironmentUserRole.objects.filter(
+                environment=environment, user=user
+            ).first()
+        ) is not None:
```

Comment:
A user can have multiple roles for an Environment.

Reply:
I thought the roles were super-sets of each other, in the sense that a developer is a ready_only role plus some additional roles. Looking closer, I see that an operator does not contain all the roles a developer has, which breaks my interpretation. 

I kind of disagree with the fact that the operator role isn't just a super-set of the developer role. The only situation that requires the user to have two roles is when they need to be an operator and a developer, which I'd argue doesn't really seem worth maintaining that relationship of multiple roles for one user in an environment